# Project 4 -- Jacqueline Sanchez 

**TA Help:** N/A

- 
    
**Collaboration:** N/A
    
- 
-

## Question 1

In [3]:
# code here
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/death_records/DeathRecords.csv")
head(myDF)

Id ResidentStatus Education1989Revision Education2003Revision
1 1  1              0                     2                    
2 2  1              0                     2                    
3 3  1              0                     7                    
4 4  1              0                     6                    
5 5  1              0                     3                    
6 6  1              0                     5                    
  EducationReportingFlag MonthOfDeath Sex AgeType Age AgeSubstitutionFlag ...
1 1                      1            M   1       87  0                   ...
2 1                      1            M   1       58  0                   ...
3 1                      1            F   1       75  0                   ...
4 1                      1            M   1       74  0                   ...
5 1                      1            M   1       64  0                   ...
6 1                      1            F   1       93  0                   ...
  CauseRecode39 NumberOfEntityAxisConditions NumberOfRecordAxisConditions Race
1 24            1                            1                            1   
2 21            3                            3                            1   
3 28            2                            2                            1   
4 22            3                            3                            1   
5 21            1                            1                            1   
6 37            5                            5                            1   
  BridgedRaceFlag RaceImputationFlag RaceRecode3 RaceRecode5 HispanicOrigin
1 0               0                  1           1           100           
2 0               0                  1           1           100           
3 0               0                  1           1           100           
4 0               0                  1           1           100           
5 0               0                  1           1           100           
6 0               0                  1           1           100           
  HispanicOriginRaceRecode
1 6                       
2 6                       
3 6                       
4 6                       
5 6                       
6 6

In [2]:
tapply(myDF$Age, myDF$Sex, mean)

F        M 
76.67668 70.22786

The mean age of death for females is 76.68 and for male is 70.23. 

## Question 2

In [3]:
# code here
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/8451/The_Complete_Journey_2_Master/5000_transactions.csv")
head(myDF)

BASKET_NUM HSHD_NUM PURCHASE_ PRODUCT_NUM SPEND UNITS STORE_R WEEK_NUM YEAR
1  24        1809     03-JAN-16 5817389     -1.50 -1    SOUTH   1        2016
2  24        1809     03-JAN-16 5829886     -1.50 -1    SOUTH   1        2016
3  34        1253     03-JAN-16  539501      2.19  1    EAST    1        2016
4  60        1595     03-JAN-16 5260099      0.99  1    WEST    1        2016
5  60        1595     03-JAN-16 4535660      2.50  2    WEST    1        2016
6 168        3393     03-JAN-16 5602916      4.50  1    SOUTH   1        2016

In [4]:
head(sort(tapply(myDF$PRODUCT_NUM, myDF$SPEND, sum)), 10)

-4.15  -4.83 118.83  -3.61  -1.85  -13.8  -6.82   -5.7  -7.43  -5.87 
  1609   6574   6871   7714   8170   8184   8186   8206   8522   8522

## Question 3

In [5]:
# code here
myDF <- fread("/anvil/projects/tdm/data/beer/reviews_sample.csv")
# head(myDF, 2)

In [6]:
# printing out the three dates that have a perfect score of 5 
tail(sort(tapply(myDF$score, myDF$date, mean)),3)

2001-04-26 2001-06-18 2002-01-26 
         5          5          5

I applied the tapply to the beer data set score column and grouped them based on dates. I used the sort function since it would sort the mean score values in ascending order the largest mean score would be placed at 

## Question 4

In [8]:
# code here
myDF <- fread("/anvil/projects/tdm/data/election/itcont1980.txt", quote = "")
names(myDF) <- c("CMTE_ID", "AMNDT_IND", "RPT_TP", "TRANSACTION_PGI", "IMAGE_NUM", "TRANSACTION_TP", "ENTITY_TP", "NAME", "CITY", "STATE","ZIP_CODE", "EMPLOYER", "OCCUPATION", "TRANSACTION_DT", "TRANSACTION_AMT", "OTHER_ID", "TRAN_ID", "FILE_NUM", "MEMO_CD", "MEMO_TEXT", "SUB_ID")
head(myDF)

CMTE_ID   AMNDT_IND RPT_TP TRANSACTION_PGI IMAGE_NUM   TRANSACTION_TP
1 C00078279 A         M11    P               80031492155 22Y           
2 C00078279 A         M11                    79031415137 15            
3 C00078279 A         M11                    79031415137 15            
4 C00078279 A         M11                    79031415137 15            
5 C00078287 A         Q1                     79031231889 15            
6 C00078287 A         Q1                     79031231889 15            
  ENTITY_TP NAME              CITY    STATE ... EMPLOYER        OCCUPATION
1 NA        MCKENNON, K R     MIDLAND MI    ...                 NA        
2 NA        OREFFICE, P       MIDLAND MI    ... DOW CHEMICAL CO NA        
3 NA        DOWNEY, J         MIDLAND MI    ... DOW CHEMICAL CO NA        
4 NA        BLAIR, E          MIDLAND MI    ... DOW CHEMICAL CO NA        
5 NA        BLANCHARD, JOHN A CHICAGO IL    ...                 NA        
6 NA        CRAMER, JOHN H    CHICAGO IL    ...                 NA        
  TRANSACTION_DT TRANSACTION_AMT OTHER_ID TRAN_ID FILE_NUM MEMO_CD
1 10031979        400                     NA      NA       NA     
2 10261979       1500                     NA      NA       NA     
3 10261979        300                     NA      NA       NA     
4 10261979       1000                     NA      NA       NA     
5  3201979        200                     NA      NA       NA     
6  2281979        200                     NA      NA       NA     
  MEMO_TEXT                      SUB_ID             
1 CONTRIBUTION REF TO INDIVIDUAL 3062020110011466469
2                                3061920110000382948
3                                3061920110000382949
4                                3061920110000382950
5                                3061920110000383914
6                                3061920110000383915

In [9]:
# storing the merged city and states into a variable "newCol" (newColumn)
newCol <- paste(myDF$CITY, myDF$STATE, sep = ",")

In [10]:
# calling the variable newCol into the main function
tail(sort(tapply(myDF$TRANSACTION_AMT, newCol, sum), n = 20))

WASHINGTON,DC LOS ANGELES,CA      DALLAS,TX     HOUSTON,TX    NEW YORK,NY 
       4273606        4569952        4748262        7606806       11345027 
             , 
      17299729

## Question 5

In [4]:
# code here
myDF <- fread("/anvil/projects/tdm/data/flights/subset/1990.csv")
head(myDF)

Year Month DayofMonth DayOfWeek DepTime CRSDepTime ArrTime CRSArrTime
1 1990 1     3          3         1707    1630       1755    1723      
2 1990 1     4          4         1706    1630       1807    1723      
3 1990 1     5          5         1629    1630       1715    1723      
4 1990 1     6          6         1633    1630       1718    1723      
5 1990 1     8          1         1630    1630       1726    1723      
6 1990 1     9          2         1734    1630       1818    1723      
  UniqueCarrier FlightNum ... TaxiIn TaxiOut Cancelled CancellationCode
1 US            29        ... NA     NA      0         NA              
2 US            29        ... NA     NA      0         NA              
3 US            29        ... NA     NA      0         NA              
4 US            29        ... NA     NA      0         NA              
5 US            29        ... NA     NA      0         NA              
6 US            29        ... NA     NA      0         NA              
  Diverted CarrierDelay WeatherDelay NASDelay SecurityDelay LateAircraftDelay
1 0        NA           NA           NA       NA            NA               
2 0        NA           NA           NA       NA            NA               
3 0        NA           NA           NA       NA            NA               
4 0        NA           NA           NA       NA            NA               
5 0        NA           NA           NA       NA            NA               
6 0        NA           NA           NA       NA            NA

In [7]:
# finding the mean departure delays on each month/dayofmonth/year triple
tail(sort(tapply(myDF$DepDelay,paste(myDF$Month,myDF$DayofMonth, myDF$Year, sep = "/"),mean, na.rm=TRUE)), 6)

12/20/1990  2/15/1990  2/16/1990 12/28/1990 12/22/1990 12/21/1990 
  27.34540   28.12302   36.19422   43.91443   45.22225   45.66178

Markdown notes and sentences and analysis written here.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.